source/particle/world.cc

```diff
@@ -852,15 +970,199 @@ namespace aspect
 
                 // Only update particles, if there are any in this cell
                 if (particles_in_cell.begin() != particles_in_cell.end())
-                  local_update_particles(cell,
-                                         particles_in_cell.begin(),
-                                         particles_in_cell.end());
+                  {
+                    local_update_particles(cell,
+                                           particles_in_cell.begin(),
+                                           particles_in_cell.end()
+#if DEAL_II_VERSION_GTE(9,3,0)
+                                           , evaluators
+#endif
+                                          );
+                  }
+
               }
         }
     }
 
 
 
+#if DEAL_II_VERSION_GTE(9,3,0)
+    namespace internal
+    {
+      template <int dim>
+      class SolutionEvaluators
```

Comment:
This class and its members have no documentation, and there is also no documentation within its member functions. Would you mind adding some?

Reply:
Separately, this class always evaluates *all* solution components. Do we have a way to state which components we really need? If not, that's ok, just wondering. (Might also not be relevant because the run time doesn't matter, but I don't really know.)

Reply:
At the moment there is no way for particle properties to signal which components they need, but yes that is a good suggestion, and if I have some time during the rest of the week I will work on that. I left a TODO. At the moment the class keeps the FEPointEvaluation members public to allow the owning object to access them individually. It is not as nice as being able to hand over a componentmask, but it works for now (the particle advection only accesses the velocity evaluator, the update uses the member functions to evaluate all components).

Reply:
Yes, that can be a long-term goal. Using `ComponentMask` is probably the right abstraction.